### PR TITLE
Add bionano

### PR DESCRIPTION
**Pull request by gallardoalba**
This PR adds the Bionano Hybrid Scaffold tool.

### CODE REVIEW

File: tools/bionano/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+categories:
+- Assembly
```
Comment (bgruening) [inline tools/bionano/.shed.yml:2]: Cris can you make a suite out of this please, in case we need to add more tools.

Comment (gallardoalba) [reply tools/bionano/.shed.yml:2]: Sure!

File: tools/bionano/bionano_scaffold_edited.xml [hunk 2]
```diff
@@ -91,7 +99,7 @@
         <flag attr="hashgen" val0="5" val1="7" val2="2.4" val3="1.5" val4="0.05" val5="5.0" val6="1" val7="1" val8="4"/>
         <flag attr="hash" val0="-hashdelta" val1="26" val2="10" val3="46"/>
         <flag attr="hashMultiMatch" val0="30" val1="10"/>
-        <flag attr="insertThreads" val0="$cores"/>
+        <flag attr="insertThreads" val0="54"/>
```
Comment (bgruening) [inline tools/bionano/bionano_scaffold_edited.xml:102]: why do you revert this?

File: tools/bionano/bionano_scaffold_edited.xml [hunk 3]
```diff
@@ -65,7 +73,7 @@
         <flag attr="RAmem" val0="3" val1="1"/>
     </global>
     <fasta2cmap>
-        <flag attr="enzyme" val0="BspQI" display="Enzyme" group="FASTA to CMAP digestion" description="Define single enzyme for in-silico FASTA to CMAP digestion. Avalible enzymes: BspQI, BbvCI, BsmI, BsrDI, BssSI, DLE1."/>
+        <flag attr="enzyme" val0="CTTAAG" display="Enzyme" group="FASTA to CMAP digestion" description="Define single enzyme for in-silico FASTA to CMAP digestion. Avalible enzymes: BspQI, BbvCI, BsmI, BsrDI, BssSI, DLE1."/>
```
Comment (bgruening) [inline tools/bionano/bionano_scaffold_edited.xml:76]: Why is this changed?

File: tools/bionano/bionano_scaffold_edited.xml [hunk 4]
```diff
@@ -178,180 +186,192 @@
...
+        <flag attr="outlier" val0="0.0001"/>
+        <flag attr="endoutlier" val0="0.001"/>
+        <flag attr="PVendoutlier"/>
+        <flag attr="deltaX" val0="12"/>
+        <flag attr="deltaY" val0="12"/>
+        <flag attr="xmapchim" val0="12"/>
+        <flag attr="hashgen" val0="5" val1="7" val2="2.4" val3="1.5" val4="0.05" val5="5.0" val6="1" val7="1" val8="4"/>
+        <flag attr="hash" val0="-hashdelta" val1="26" val2="10" val3="46"/>
+        <flag attr="hashMultiMatch" val0="30" val1="10"/>
+        <flag attr="insertThreads" val0="54"/>
```
Comment (bgruening) [inline tools/bionano/bionano_scaffold_edited.xml:203]: why this? we should use $cores

File: tools/bionano/bionano_scaffold_edited.xml [hunk 5]
```diff
@@ -420,6 +456,15 @@
                     <has_archive_member path=".*/status.txt"/>
                 </assert_contents>
             </output>
+                        <assert_stdout>
```
Comment (bgruening) [inline tools/bionano/bionano_scaffold_edited.xml:459]: indentation? Are you not using the galaxy-language-server?

File: tools/bionano/bionano_scaffold_edited.xml [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+        #if $configuration_options.configuration == 'vgp'
+            && cp '${vgp_mode}' ./config.xml
+        #else
+            #set cores = os.environ.get('GALAXY_SLOTS', 2)
+            #set memory = os.environ.get('GALAXY_MEMORY_MB', 4000) / 1000
+            && cp '${configuration_file}' ./config.xml
+            && sed -i 's|attr="maxmem" val0=.* display|attr="maxmem" val0="$memory" display|' ./config.xml
+            && sed -i 's|attr="maxthreads" val0=.* display|attr="maxthreads" val0="$cores" display|' ./config.xml
+            && sed -i 's|attr="maxvirtmem" val0=.*/>|attr="maxvirtmem" val0="$cores"/>|' ./config.xml
+            && cat ./config.xml
```
Comment (bgruening) [inline tools/bionano/bionano_scaffold_edited.xml:23]: this cat should happen in both cases.... and then we can have tests that check that the core settings are correct

File: tools/bionano/bionano_scaffold_edited.xml (hunk 2 above)
Comment (gallardoalba) [reply tools/bionano/bionano_scaffold_edited.xml:102]: Yes, sorry, It was a fail. I changed all the configuration file because the one provided in the test correspond to the samples used on it. I'll fix it.

Comment (gallardoalba) [reply tools/bionano/bionano_scaffold_edited.xml:102]: I didn't realize that there were more parameters that required to be modified.

Comment (bgruening) [reply tools/bionano/bionano_scaffold_edited.xml:102]: There are a few more ...

File: tools/bionano/bionano_scaffold_edited.xml (hunk 3 above)
Comment (gallardoalba) [reply tools/bionano/bionano_scaffold_edited.xml:76]: Because the VGP configuration uses that sequence (corresponding to a specific restriction enzyme); the input files used in the test uses a different one.

File: tools/bionano/bionano_scaffold_edited.xml (hunk 4 above)
Comment (gallardoalba) [reply tools/bionano/bionano_scaffold_edited.xml:203]: Yes, I'll fix it.

File: tools/bionano/bionano_scaffold_edited.xml (hunk 6 above)
Comment (gallardoalba) [reply tools/bionano/bionano_scaffold_edited.xml:23]: The problem is that the VGP configuration file cannot be used with the input files used in the tests because the restriction enzymes are different.

File: tools/bionano/bionano_scaffold_edited.xml (hunk 3 above)
Comment (bgruening) [reply tools/bionano/bionano_scaffold_edited.xml:76]: ok, maybe that parameter should be made configureable?

Comment (gallardoalba) [reply tools/bionano/bionano_scaffold_edited.xml:76]: I think that It wouldn't be necessary (except for testing purposes). But if you consider so I'll include it.

File: tools/bionano/bionano_scaffold_edited.xml (hunk 5 above)
Comment (gallardoalba) [reply tools/bionano/bionano_scaffold_edited.xml:468]: Yes, I just copied this block from the one which you included.

File: tools/bionano/bionano_scaffold_edited.xml [hunk 7]
```diff
@@ -76,7 +75,7 @@
         <flag attr="RAmem" val0="3" val1="1"/>
     </global>
     <fasta2cmap>
-        <flag attr="enzyme" val0="CTTAAG" display="Enzyme" group="FASTA to CMAP digestion" description="Define single enzyme for in-silico FASTA to CMAP digestion. Avalible enzymes: BspQI, BbvCI, BsmI, BsrDI, BssSI, DLE1."/>
+        <flag attr="enzyme" val0="${configuration_options.enzyme}" display="Enzyme" group="FASTA to CMAP digestion" description="Define single enzyme for in-silico FASTA to CMAP digestion. Avalible enzymes: BspQI, BbvCI, BsmI, BsrDI, BssSI, DLE1."/>
```
Comment (gallardoalba) [inline tools/bionano/bionano_scaffold_edited.xml:78]: This is not possible because the $configuration_options.enzyme is only set when  $configuration_options == 'VGP'

Comment (bgruening) [reply tools/bionano/bionano_scaffold_edited.xml]: Than add an if-condition.

File: tools/bionano/bionano_scaffold_edited.xml [hunk 8]
```diff
@@ -0,0 +1,614 @@
...
+            <param name="conflict_filter_genome" value="3"/>
+            <param name="conflict_filter_sequence" value="3"/>
+            <conditional name="configuration_options">
+                <param name="configuration" value="file"/>
+                <param name="configuration_file" value="configuration.xml"/>
+            </conditional>
+            <param name="zip_file" value="true"/>
+            <output name="ngs_contigs_scaffold_fasta" ftype="fasta">
+                <assert_contents>
+                    <has_size value="4753369" delta="300" />
```
Comment (bgruening) [inline tools/bionano/bionano_scaffold_edited.xml:445]: you can test a FASTA file in a smarter way than size. E.g you can add line count, assert some header line etc in addition.

File: tools/bionano/bionano_scaffold_edited.xml [hunk 9]
```diff
@@ -0,0 +1,614 @@
...
+            <param name="bionano_cmap" value="colormap_assembly.cmap"/>
+            <param name="conflict_filter_genome" value="2"/>
+            <param name="conflict_filter_sequence" value="2"/>
+            <conditional name="configuration_options">
+                <param name="configuration" value="file"/>
+                <param name="configuration_file" value="configuration.xml"/>
+            </conditional>
+            <output name="ngs_contigs_scaffold_fasta" ftype="fasta">
+                <assert_contents>
+                    <has_size value="4753369" delta="300" />
```
Comment (bgruening) [inline tools/bionano/bionano_scaffold_edited.xml:484]: same here

**Comment by bgruening** [comment]
Cool, do you add this to EU?
